src/algorithms/highlevel/crosssimilaritymatrix.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2006-2016  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+#include "crosssimilaritymatrix.h"
+#include "essentiamath.h"
+#include <vector>
+#include <iostream>
+#include <string>
+#include <algorithm>
+#include <functional>
+
+namespace essentia {
+namespace standard {
+
+const char* CrossSimilarityMatrix::name = "CrossSimilarityMatrix";
+const char* CrossSimilarityMatrix::category = "Music Similarity";
+const char* CrossSimilarityMatrix::description = DOC("This algorithm computes a euclidean cross-similarity matrix of two sequences of frame features. Similarity values can be optionally binarized\n\n"
+"Use default parameter values for best results while computing cross-similarity using the binarize method.\n\n"
```

Comment:
These parameters are good for cover detection, not the other tasks, so mentions that instead (something like "... the default parameters for binarizing are optimized according to [reference] for cover song identification using chroma features..." ).